src/smart-components/access-management/AddUserGroupModal.tsx

```diff
@@ -0,0 +1,60 @@
+import { Button, Modal } from '@patternfly/react-core';
+import React from 'react';
+import UserGroupsTable from './UserGroupsTable';
+import { useDispatch } from 'react-redux';
+import { addMembersToGroup } from '../../redux/actions/group-actions';
+
+interface AddUserGroupModalProps {
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+  selectedUsers: any[];
+}
+
+export const AddUserGroupModal: React.FunctionComponent<AddUserGroupModalProps> = ({ isOpen, setIsOpen, selectedUsers }) => {
+  const [selectedGroups, setSelectedGroups] = React.useState<any[]>([]);
+  const handleUserGroupsChange = (groups: any[]) => setSelectedGroups(groups);
+  const dispatch = useDispatch();
+
+  const handleCloseModal = () => setIsOpen(false);
+
+  const handleAddUsers = () => {
+    const selectedUsernames = selectedUsers.map((user) => ({username: user[0]})); // TODO: fix - this seems gross
+    selectedGroups.forEach((group) => {
+      console.log(`Adding ${JSON.stringify(selectedUsernames)} to group ${group.name} - ${group.uuid}`);
+      //dispatch(addMembersToGroup(group.uuid, selectedUsernames)); // TODO: fix 'user' not found 404 error
```

Comment:
I wasn't able to get 404 locally, but I got 400 - we should not try to update system roles, these have to be excluded from the list or disabled (might be better to check with UX what is preferred). Please, let me know if you still have issues figuring that out